editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx

```diff
@@ -6,7 +6,7 @@ import React, { createContext, useContext, useEffect, useState } from "react";
 
 import { Store, useStore } from "./store";
 
-export type AnalyticsType = "init" | "resume";
+export type AnalyticsType = "init" | "resume" | "reset"
```

Comment:
Hey @jessicamcinchak something I hadn't realised is that this `AnalyticsType` is related to the `analytics` model which tracks the overarching analytics session? It stores whether the user's session was an `init` or a `resume`?

I believe the log direction only applies to `analytics_log` model instances which are tied together with an instance of the `analytics` model?

Based on the way the `"reset"` is being recorded against an `analytics_log` if it make more sense to think of it as a log direction i.e. an extreme case of going `backwards` where you go back to the the start?

Reply:
I'm not entirely sure I'm following this one yet, but my instinct here is that "reset" _ends the given session_ and therefore, at least in my mind, it's an analytics type rather than pure "direction" (eg "backwards" is always _within_ a session). But also totally trust your judgement on this!

Reply:
I think that's a really good point although in this PR the `"reset"` is only ever applied within a session as a `flow_direction`

As in:
- The last `analytics_log` for that `analytic` session would have a `flow_direction` of `"reset"`. 
- The `type` of the aforementioned `analytic` session would remain whatever `type` it had been i.e. `"init"` or `"resume"`
- The new `analytic` session would default to `"init"`

So currently the `AnalyticType` is only applied as a `flow_direction`.  

Does that make sense? I'm maybe over focussing on the semantics but I'm just trying to get my head around who things have been working and what we're looking for with this new functionality.

Reply:
That's a very helpful framing of the situation, thanks Jess! 

I'll have a look at "Where applicants drop-off" and see how it's being approached there and if there are any implications with this change.

Reply:
I had a wee look at the SQL query:

```
SELECT date_trunc('week', analytics_logs.created_at) as "Week", count(*) as "Exits"
FROM analytics_logs
JOIN analytics ON (analytics.id = analytics_logs.analytics_id)
WHERE analytics_logs.user_exit = true
AND analytics_logs.node_type = 3
AND analytics.flow_id = {{flow_id}}::uuid
GROUP BY "Week"
ORDER BY "Week"
```

The `drop-off` uses the `user_exit` property which to my understanding is marked as `true` when a user stops viewing the page:

https://github.com/theopensystemslab/planx-new/blob/a6362685baeb3b3c5068cde3bb499dcc50d6ffc4/editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx#L60-L86

I had thought of a user deciding to `reset` as not actually exiting the page and therefore not being a `user_exit: true`. 

Although in the context of drop off I'm not so sure 🤔

Reply:
Actually on second thoughts I think we'll be able to handle the `reset` state in a similar fashion as the query above so we should be okay I think 👍